linux/iscsi.go

```diff
@@ -300,43 +322,33 @@ func isReachable(initiatorIP, targetIP string) (reachable bool, err error) {
 	return true, nil
 }
 
-func updateConnectionMode(target *model.IscsiTarget, targets model.IscsiTargets, connectionMode string) {
+func updateConnectionMode(target *model.IscsiTarget, connectionMode string) {
 	log.Tracef(">>>>> updateConnectionMode for target %s mode %s", target.Name, connectionMode)
 	defer log.Trace("<<<<< updateConnectionMode")
 
-	iscsiMutex.Lock()
-	defer iscsiMutex.Unlock()
-
 	if !(connectionMode == "automatic" || connectionMode == "manual") {
 		err := fmt.Errorf("unsupported iscsi connection mode %s", connectionMode)
 		log.Debug(err.Error())
 		return
 	}
 
-	for _, t := range targets {
-		if t.Name != target.Name {
-			continue
-		}
-		log.Tracef("updating node.startup for target %s address %s", t.Name, t.Address)
-		// update connection mode of all targets with same target name
-		args := []string{"--mode", "node", "--targetname", t.Name, "--portal", t.Address, "--op", "update", "-n", "node.startup", "-v", connectionMode}
-		_, _, err := util.ExecCommandOutput(iscsicmd, args)
-		if err != nil {
-			err = fmt.Errorf("Unable to update node.startup to %s for node %s error %s", connectionMode, t.Name, err.Error())
-			log.Errorf(err.Error())
-		}
+	log.Tracef("updating node.startup for target %s address %s", target.Name, target.Address)
+	// update connection mode of all targets with same target name
+	args := []string{"--mode", "node", "--targetname", target.Name, "--portal", target.Address, "--op", "update", "-n", "node.startup", "-v", connectionMode}
+	_, _, err := util.ExecCommandOutput(iscsicmd, args)
+	if err != nil {
+		err = fmt.Errorf("Unable to update node.startup to %s for node %s error %s", connectionMode, target.Name, err.Error())
+		log.Errorf(err.Error())
 	}
 	return
 }
 
 // updates iscsi node db with given chap username
 func updateChapUser(target *model.IscsiTarget, chapUser string) (err error) {
-	log.Tracef(">>>>> updateChapUser for target %s user %s", target.Name, chapUser)
+	log.Tracef(">>>>> updateChapUser for target %s portal %s, user %s", target.Name, target.Address, chapUser)
 	defer log.Trace("<<<<< updateChapUser")
 
-	iscsiMutex.Lock()
```

Comment:
lets keep these, they will prevent iscsi db corruption.

Reply:
they cause nested lock exception since addTarget takes the lock before this

Reply:
ok..got it..iscsi mutex is held there..